### DISCUSSION TITLE
Passing secrets as inputs to composite actions. Is it safe ?

### DISCUSSION THREAD

**Discussion by rekaparathalwar** [community/community#34212] [Actions]
Composite action doesnt support passing secrets. Instead we can pass the secrets as inputs to composite action.

Is it safe doing so ?  Is there any security implications ?

**Comment by airtower-luna** [comment]
The security implications are the same as for passing secrets to any other type of action: You're giving the secret to the action, and potentially (depending on how the action handles its inputs) any code the action calls. For a composite action that might include other actions it calls.

To see if the handling is safe you'll have to audit the specific action, and anything it calls with access to the secret input. If you determine the action is safe you might want to reference it by commit ID in your workflow to make sure it can't change without you noticing.

**Reply by jsugarman** [reply]
I have a couple questions on this
- Assuming the action is handling the secret safely is there any risk of inputs that are secrets being logged in plain text by the github action?
-  Also, what, then, is the purpose of the [secrets option for reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow) in that case? Is it [registering the secret](https://github.com/actions/toolkit/blob/main/packages/core/README.md#setting-a-secret) input in some way

**Reply by madkins-webscale** [reply]
From my testing it appears they remain redacted in the logs even after being passed across to a composite action as an input (and referred to then via the inputs context)

**Reply by alexmercerind** [reply]
i wasted so much time now
